test/utils/base_elements/base_collection.js

```diff
@@ -0,0 +1,29 @@
+const logger = require('../../../test/config/logger.config.js');
+
+class Collection {
+    constructor(elementName, selector) {
+        this.collection = element.all(by.css(selector));
+        this.elementName = elementName;
+    };
+    async getCount() {
+        const collectionCount = await this.collection.count();
+        logger.info(`Count of "${this.elementName}" is "${collectionCount}"`); 
+        return collectionCount;
+    };
+    async getTexts() {
+        const arrayOfCollectionTexts = await this.collection.getText();
+        logger.info(`Texts of "${this.elementName}" are [${arrayOfCollectionTexts}]`);
+        return arrayOfCollectionTexts;
+    };
+    async clickElementByText(textToClick) {
+        const arrayOfElementTexts = await this.collection.getText();
+        const elementToClickIndex = arrayOfElementTexts.indexOf(textToClick);
+        if (elementToClickIndex === -1) {
+            throw new Error(`No element with [${textToClick}] text found!`);
+        }
+        logger.info(`Clicking "${textToClick}" text in "${this.elementName}"`);
+        return this.collection.get(elementToClickIndex).click();
```

Comment:
There is no need to return this statement.

Reply:
fixed